hcl/eval/object.go

```diff
@@ -151,6 +153,23 @@ func (obj *Object) SetAt(path ObjectPath, value Value) error {
 		path = path[1:]
 	}
 
+	old, ok := obj.GetKeyPath(path)
+	if ok {
+		// TODO(i4k): document this properly.
```

Comment:
On top of documenting that this is about parent detection we also need to make it clear that origin always must be inside the hierarchy of a stack or else this will not work (like the other bug related to the origin being from the import block file, outside the hierarchy).

Another idea I had at the time when I was thinking on something similar is to keep the hierarchical level of values, not the origin itself, so the object API would be more explicit about the hierarchical concept and how it affects behavior, but Im not sure it is a good idea, just sharing one of the ideas I had.